flexget/plugins/notifiers/notify.py

```diff
@@ -2,208 +2,67 @@
         """
-        Tries to render a template, fallback to default template and just value if unsuccessful
-
-        :param entity: The entity to operate on, either `Entry` or `Task`
-        :param data: The text to be rendered
-        :param attribute: Attribute name to be fetched from the defaults
-        :param default_dict: The default dict, depending on entity type
-        :return: A rendered value or original value
+        Send a notification out to the given `notifiers` with a given `title` and `message`.
+        If `template_renderer` is specified, `title`, `message`, as well as any string options in a notifier's config
+        will be rendered using this function before sending the message.
+
+        :param str title: Title of the notification.
+        :param str message: Body of the notification.
+        :param list notifiers: A list of configured notifier output plugins.
+        :param template_renderer: A function that should be used to render any jinja strings in the configuration.
         """
-        result = data
-
-        # Handles file templates
-        if attribute == 'file_template':
+        if template_renderer:
             try:
-                data = get_template(data, scope)
-            except ValueError as e:
-                log.warning(e.args[0])
-                return
-
-        try:
-            result = entity.render(data)
-        except (RenderError, ValueError) as e:
-            log.debug('cannot render: `%s: %s`, error: %s', attribute, data, e.args[0])
-            if attribute in default_dict:
-                try:
-                    log.debug('trying to render default value for `%s`', attribute)
-                    result = entity.render(default_dict[attribute])
-                # Render error on defaults should not happen
-                except RenderError as e:
-                    log.warning('default dict failed to render: `%s: %s`. Error: %s. Reverting to original value.',
-                                attribute, data, e.args[0])
-        else:
-            # Even in debug level, showing contents of rendered file templates is super spammy
-            if attribute == 'file_template':
-                message = 'successfully rendered file_template %s', data
-            else:
-                message = 'successfully rendered `%s` to %s', attribute, result
-            log.debug(*message)
-        return result
-
-    def send_notification(self, task, config):
-        config = self.prepare_config(config)
-        scope = config.pop('scope')
-        what = config.pop('what')
-        notifiers = config.pop('to')
-
-        # Creating iterator based on config scope
-        iterate_on = [getattr(task, container) for container in what] if scope == 'entries' else [[task]]
-
+                title = template_renderer(title)
+            except RenderError as e:
+                log.error('Error rendering notification title: %s', e)
```

Comment:
I sorta feel like having the unrendered string will be a better indication of error when debugging why their config isn't working rather than getting some string that isn't to do with the title/message they defined.